ntp-daemon/src/peer_manager.rs

```diff
@@ -124,8 +170,54 @@ impl<C: NtpClock> Peers<C> {
         )
     }
 
-    pub async fn add_peer(&mut self, config: PeerConfig) -> JoinHandle<()> {
-        self.add_peer_internal(Arc::new(config)).await
+    async fn add_peer_internal(&mut self, config: PeerConfig) -> Vec<JoinHandle<()>> {
+        let new_active = match &config {
+            PeerConfig::Standard(StandardPeerConfig { addr, .. }) => {
+                // unwrap is safe because of the peek() in `socket_addresses`
+                let address = socket_addresses(addr).await.next().unwrap();
+
+                vec![address]
+            }
+            PeerConfig::Pool(PoolPeerConfig { hosts, max_peers }) => {
+                let pool_status = self.pools.entry(hosts.clone()).or_default();
```

Comment:
Why is it not a problem if the config doesn't match to a pool, that seems like a critical failure to me?